modules/m_token_auth.h

```diff
@@ -0,0 +1,327 @@
+
+	AuthToken* FindToken(const Anope::string& token);
+
+	AuthToken* GetToken(const Anope::string& name) const;
+
+	AuthToken* GetToken(unsigned pos) const;
+
+	void Clear();
+
+	void UpdateToken(const Serialize::Reference<AuthToken>& token);
+
+	bool IsEmpty() const
+	{
+		return tokens.empty();
+	}
+
+	unsigned int GetSize()
+	{
+		return list.size();
+	}
+};
+
+ExtensibleRef<AuthTokenList> GetTokenExtRef()
+{
+	ExtensibleRef<AuthTokenList> extRef(TOKENS_EXT_NAME);
+	return extRef;
+}
+
+AuthTokenList* GetTokenList(NickCore* nc, bool create = false)
+{
+	ExtensibleRef<AuthTokenList> extRef(TOKENS_EXT_NAME);
+	if (!extRef)
+		return NULL;
+
+	if (create)
+		return extRef->Require(nc);
+
+	return extRef->Get(nc);
+}
+
+
+void AuthTokenList::Clear()
+{
+	for (tokenlist::iterator it = list.begin(); it != list.end();)
+	{
+		tokenlist::iterator oldit = it++;
+		delete *oldit;
+	}
+}
+
+void AuthTokenList::UpdateToken(const Serialize::Reference<AuthToken>& token)
+{
+	this->DelToken(token);
+	this->AddToken(token);
+}
+
+AuthToken* AuthTokenList::GetToken(unsigned pos) const
+{
+	if (pos >= list.size())
+		return NULL;
+
+	return list[pos];
+}
+
+AuthToken* AuthTokenList::GetToken(const Anope::string& name) const
+{
+	tokenmap::const_iterator it = name_map.find(name);
+	if (it == name_map.end())
+		return NULL;
+
+	return it->second;
+}
+
+void AuthTokenList::DelToken(AuthToken* token)
+{
+	tokens.erase(token->GetToken());
+	name_map.erase(token->GetName());
+	tokenlist::iterator it = std::find(list.begin(), list.end(), token);
+	if (it != list.end())
+		list.erase(it);
+}
+
+AuthToken* AuthTokenList::FindToken(const Anope::string& token)
+{
+	tokenmap::iterator it = tokens.find(token);
+	if (it == tokens.end())
+		return NULL;
+
+	return it->second;
+}
+
+bool AuthTokenList::AddToken(AuthToken* token)
+{
+	if (!tokens.insert(std::make_pair(token->GetToken(), token)).second)
+		return false;
+
+	if (!name_map.insert(std::make_pair(token->GetName(), token)).second)
+	{
+		tokens.erase(token->GetToken());
+		return false;
+	}
+	list.push_back(token);
+	return true;
+}
+
+AuthToken* AuthTokenList::NewToken(const Anope::string& name)
+{
+	AuthToken* t = new AuthToken(nc, name, Anope::Random(TOKEN_LEN));
+
+	if (!this->AddToken(t))
+	{
+		delete t;
+		return NULL;
+	}
+
+	t->QueueUpdate();
+
+	return t;
+}
+
+Serializable* AuthToken::Unserialize(Serializable* obj, Serialize::Data& data)
+{
+	ExtensibleRef<AuthTokenList> extRef(TOKENS_EXT_NAME);
+	Anope::string disp, name, token;
+	time_t settime;
+
+	data["nc"] >> disp;
+	data["name"] >> name;
+	data["token"] >> token;
+	data["set_at"] >> settime;
+
+	NickCore* nc = NickCore::Find(disp);
+	if (!nc)
+		return NULL;
+
+	AuthToken* tkn;
+
+	if (obj)
+	{
+		tkn = anope_dynamic_static_cast<AuthToken*>(obj);
+		if (name != tkn->GetName())
+			tkn->SetName(name);
+
+		if (token != tkn->GetToken())
+			tkn->SetToken(token);
+
+		tkn->set_at = settime;
+	}
+	else
+		tkn = new AuthToken(nc, name, token, settime);
```

Comment:
If you brace the if the else should have braces too for clarity
```suggestion
	{
			tkn = new AuthToken(nc, name, token, settime);
	}
```